profiler/shakesapp/shakesapp/client.go

```diff
@@ -52,8 +52,8 @@ func SimulateClient(ctx context.Context, addr string, numReqs, reqsInFlight int)
 			inFlightCh <- true
 			q := queries[rand.Intn(len(queries))]
 			resp, err := client.GetMatchCount(ctx, &ShakespeareRequest{Query: q.query})
-			if resp.MatchCount != q.wantMatchCount {
-				panic(fmt.Sprintf("GetMatchCount(%q): got %d matches, want %d", q.query, resp.MatchCount, q.wantMatchCount))
+			if err == nil && (resp.MatchCount != q.wantMatchCount) {
```

Comment:
I think

```
inFlightCh <- true
respErrs <- func() error {
  q := queries[rand.Intn(len(queries))]
  resp, err := client.GetMatchCount(ctx, &ShakespeareRequest{Query: q.query})
  if err != nil {
    return err
  }
  if resp.MatchCount != q.wantMatchCount {
    return fmt.Errorf("GetMatchCount(%q): got %d matches, want %d", q.query, resp.MatchCount, q.wantMatchCount)
  }
}()
<-inFlightCh
```

might be a bit more convenient to read in terms of how control flow is usually structured in Go programs ("if err != nil { return err }").

Reply:
Done.